backend/src/Controller/PlantController.php

```diff
@@ -38,8 +35,32 @@ public function createPlant(Request $request, PlantRepository $plantRepository,
             );
         }
         $plantRepository->savePlant($plant);
+        return $this->jsonResponse($plant, $serializer);
+    }
+
+    /**
+     * @Route("/plant/{id}", methods={"PUT"})
+     */
+    public function updatePlant($id, SerializerInterface $serializer, PlantRepository $plantRepository, Request $request, ValidatorInterface $validator): JsonResponse {
+        $plant = $plantRepository->findOneBy(array('id' => $id));
+        $newData = $serializer->deserialize($request->getContent(), Plant::class, 'json');
+        $validationResult = $validator->validate($newData);
+        if ($validationResult->count() !== 0) {
+            return new JsonResponse(
+                ["error" => "Plant data invalid."],
+                JsonResponse::HTTP_BAD_REQUEST
+            );
+        }
+
+        $validatedData = json_decode($serializer->serialize($newData, 'json'), true);
```

Comment:
wieso?

Reply:
das sollte doch eine Plant sein und kein stdClass

Reply:
plant in currentPlant und newData in newPlant, alright :)